lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -25,6 +25,62 @@
 
 using namespace swift;
 
+bool swift::hasEscaped(BorrowedValue value) {
+  GraphNodeWorklist<Operand *, 8> worklist;
+  for (Operand *use : value->getUses()) {
+    if (use->getOperandOwnership() != OperandOwnership::NonUse)
+      worklist.insert(use);
+  }
+
+  while (Operand *op = worklist.pop()) {
+    switch (op->getOperandOwnership()) {
+    case OperandOwnership::NonUse:
+    case OperandOwnership::TrivialUse:
+    case OperandOwnership::ForwardingConsume:
+    case OperandOwnership::DestroyingConsume:
+      llvm_unreachable("this operand cannot handle an inner guaranteed use");
+
+    case OperandOwnership::ForwardingUnowned:
+    case OperandOwnership::PointerEscape:
+    case OperandOwnership::BitwiseEscape:
```

Comment:
We should treat `BitwiseEscape` as a normal instantaneous use.
Eventually, we should do the same for `ForwardingUnowned` but we should do that in canonicalization first before handling it optimistically here.